filetype/types/document.py

```diff
@@ -0,0 +1,266 @@
+            )
+        if self.compare_bytes(buf, b"ppt/", offset):
+            return (
+                self.mime
+                == "application/vnd.openxmlformats-officedocument.presentationml.presentation"
+            )
+        if self.compare_bytes(buf, b"xl/", offset):
+            return (
+                self.mime
+                == "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
+            )
+
+
+class Doc(Type):
+    """
+    Implements the Microsoft Word (Office 97-2003) document type matcher.
+    """
+
+    MIME = "application/msword"
+    EXTENSION = "doc"
+
+    def __init__(self):
+        super(Doc, self).__init__(mime=Doc.MIME, extension=Doc.EXTENSION)
+
+    def match(self, buf):
+        header_match = (
+            len(buf) > 8 and buf[0:8] == b"\xD0\xCF\x11\xE0\xA1\xB1\x1A\xE1"
+        )
+        subheader_match = (
+            header_match
+            and len(buf) > 515
+            and (
+                buf[512:515] == b"\xEC\xA5\xC1\x00"
+                or (
+                    len(buf) > 2142
+                    and b"\x00\x0A\x00\x00\x00MSWordDoc\x00\x10\x00\x00\x00Word.Document.8\x00\xF49\xB2q"
+                    in buf[2075:2142]
+                )
+            )
+        )
+        return header_match and subheader_match
+
+
+class Docx(OfficeOpenXml):
+    """
+    Implements the Microsoft Word OOXML (Office 2007+) document type matcher.
+    """
+
+    MIME = "application/vnd.openxmlformats-officedocument.wordprocessingml.document"
+    EXTENSION = "docx"
+
+    def __init__(self):
+        super(Docx, self).__init__(mime=Docx.MIME, extension=Docx.EXTENSION)
+
+
+class Odt(OpenDocument):
+    """
+    Implements the OpenDocument Text document type matcher.
+    """
+
+    MIME = "application/vnd.oasis.opendocument.text"
+    EXTENSION = "odt"
+
+    def __init__(self):
+        super(Odt, self).__init__(mime=Odt.MIME, extension=Odt.EXTENSION)
+
+
+class Xls(Type):
+    """
+    Implements the Microsoft Excel (Office 97-2003) document type matcher.
+    """
+
+    MIME = "application/vnd.ms-excel"
+    EXTENSION = "xls"
+
+    def __init__(self):
+        super(Xls, self).__init__(mime=Xls.MIME, extension=Xls.EXTENSION)
+
+    def match(self, buf):
+        header_match = (
+            len(buf) > 8 and buf[0:8] == b"\xD0\xCF\x11\xE0\xA1\xB1\x1A\xE1"
+        )
+        subheader_match = (
+            header_match
+            and len(buf) > 520
+            and (
+                (
+                    buf[512:516] == b"\xFD\xFF\xFF\xFF"
+                    and (buf[518] == 0x00 or buf[518] == 0x02)
+                )
+                or (buf[512:520] == b"\x09\x08\x10\x00\x00\x06\x05\x00")
+                or (
+                    len(buf) > 2095
+                    and b"\xE2\x00\x00\x00\x5C\x00\x70\x00\x04\x00\x00Calc"
+                    in buf[1568:2095]
+                )
+            )
+        )
+        return header_match and subheader_match
```

Comment:
```suggestion
        if len(buf) > 520 and buf[0:8] == b"\xD0\xCF\x11\xE0\xA1\xB1\x1A\xE1":
            if buf[512:516] == b"\xFD\xFF\xFF\xFF" and (buf[518] == 0x00 or buf[518] == 0x02) \
                or buf[512:520] == b"\x09\x08\x10\x00\x00\x06\x05\x00":
                return True
            elif len(buf) > 2095:
                return b"\xE2\x00\x00\x00\x5C\x00\x70\x00\x04\x00\x00Calc" in buf[1568:2095]
        return False
```
Please, check my suggestion. May be spaces missed/extra.